lib/iris/cube.py

```diff
@@ -2173,8 +2194,10 @@ def new_cell_measure_dims(cm_):
         dimension_mapping, data = iris.util._slice_data_with_keys(
             cube_data, keys)
 
-        # We don't want a view of the data, so take a copy of it.
-        data = deepcopy(data)
+        # We don't want a view of the data, so take a copy of it, unless
+        # self.share_data is True.
+        if not self.share_data:
+            data = deepcopy(data)
```

Comment:
I'm concerned that for such a large change to cube behaviour the testing is very light. We have no idea at all what will happen in general use cases – at the very least this change requires some integration testing, but some more robust unit testing as well would also be good.